### PR TITLE
feat(vue): add support vue devtools options

### PR DESCRIPTION
**Pull request by niklas-wortmann**
## Changes

- Vue Devtools support a launchEditor for "open in Editor"
- This accepts the option and forwards it to the vue dev tools

## Testing

- no Idea how :D 
- more than happy to add a test with some guidance

## Docs

Would probably make sense to update this section in the docs, happy to file a separate pr
/cc @withastro/maintainers-docs for feedback!

https://github.com/withastro/docs/pull/8295

### CODE REVIEW

File: packages/integrations/vue/src/index.ts [hunk 1]
```diff
@@ -125,9 +126,11 @@ async function getViteConfiguration(
 
 	if (command === 'dev' && options?.devtools) {
 		const vueDevTools = (await import('vite-plugin-vue-devtools')).default;
+		const launchEditor =  typeof options.devtools === 'object' ? options.devtools.launchEditor : "code"
 		config.plugins?.push(
 			vueDevTools({
 				appendTo: VIRTUAL_MODULE_ID,
+				launchEditor
```
Comment (webfansplz) [inline packages/integrations/vue/src/index.ts:133]: I suggest we directly introduce into the whole ` VitePluginVueDevToolsOptions` (except ` appendTo `),  like:

```
	vueDevTools({
        ...devtoolsOptions,
				appendTo: VIRTUAL_MODULE_ID,
 })
```
This aligns with the plugin's own configuration, making it more intuitive for users and easier to extend.

File: packages/integrations/vue/src/index.ts [hunk 2]
```diff
@@ -5,14 +5,15 @@ import type { Options as VueJsxOptions } from '@vitejs/plugin-vue-jsx';
...
+import type {VitePluginVueDevToolsOptions} from "vite-plugin-vue-devtools";
 
 const VIRTUAL_MODULE_ID = 'virtual:@astrojs/vue/app';
 const RESOLVED_VIRTUAL_MODULE_ID = `\0${VIRTUAL_MODULE_ID}`;
 
 interface Options extends VueOptions {
 	jsx?: boolean | VueJsxOptions;
 	appEntrypoint?: string;
-	devtools?: boolean;
+	devtools?: boolean  | {launchEditor: VitePluginVueDevToolsOptions['launchEditor'];};
```
Comment (florian-lefebvre) [inline packages/integrations/vue/src/index.ts:16]:
```suggestion
	devtools?: boolean  | Omit<VitePluginVueDevToolsOptions, "appendTo">;
```

File: packages/integrations/vue/src/index.ts [hunk 3]
```diff
@@ -125,9 +126,11 @@ async function getViteConfiguration(
 
 	if (command === 'dev' && options?.devtools) {
 		const vueDevTools = (await import('vite-plugin-vue-devtools')).default;
+		const launchEditor =  typeof options.devtools === 'object' ? options.devtools.launchEditor : "code"
```
Comment (florian-lefebvre) [inline packages/integrations/vue/src/index.ts:129]:
```suggestion
		const pluginOptions =  typeof options.devtools === 'object' ? options.devtools.launchEditor : {}
```

File: packages/integrations/vue/src/index.ts (hunk 1 above)
Comment (florian-lefebvre) [inline packages/integrations/vue/src/index.ts:133]:
```suggestion
				...pluginOptions,
				appendTo: VIRTUAL_MODULE_ID,
```

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]
Thanks a lot! A few things:
- I added a few suggestions after @webfansplz comment
- Testing manually is definitely fine!
- Don't forget to add a changeset by running `pnpm changeset` (should be a minor I think)
- It will require a docs PR, feel free to ping me on the associated PR!

**Comment by niklas-wortmann** [comment]
I wasn't sure about the changeset thing, other than that it should be fine now

**Comment by florian-lefebvre** [comment]
Awesome thanks! I updated the changeset, can you make a docs PR? You can inspiration from the changeset if that helps you!

**Review by florian-lefebvre — APPROVED** [review]

**Comment by Princesseuh** [comment]
Seems like something happened in the lockfile, could you revert that change?

File: packages/integrations/vue/src/index.ts [hunk 4]
```diff
@@ -125,9 +126,11 @@ async function getViteConfiguration(
 
 	if (command === 'dev' && options?.devtools) {
 		const vueDevTools = (await import('vite-plugin-vue-devtools')).default;
+		const pluginOptions =  typeof options.devtools === 'object' ? options.devtools : {}
```
Comment (Princesseuh) [inline packages/integrations/vue/src/index.ts:129]:
```suggestion
		const devToolsOptions =  typeof options.devtools === 'object' ? options.devtools : {}
```
perhaps? I was confused what this was about reading the PR.

**Comment by niklas-wortmann** [comment]
Thanks @florian-lefebvre for cleaning up after me 😬

**Review by Princesseuh — APPROVED** [review]

**Pull request merged**